src/api/test/functional/source_services_test.rb

```diff
@@ -260,6 +247,64 @@ def test_run_source_service
     assert_response 404
   end
 
+  def test_service_merge
+    login_tom
+    raw_put '/source/home:tom/service/_meta', "<package project='home:tom' name='service'> <title /> <description /> </package>"
+    assert_response :success
+    raw_put '/source/home:tom/service/pack.spec', "# Comment \nName: pack\nVersion: 12\nRelease: 9\nSummary: asd"
+    assert_response :success
+
+    raw_put '/source/home:tom/service/_service', '<services> <service name="not_existing" /> </services>'
```

Comment:
I'd suggest to split this up in several test cases. E.g.

1) service merge failed
2) service merge successful
etc